invenio_files_rest/views.py

```diff
@@ -267,6 +267,16 @@ def inner(self, bucket, key, upload_id, *args, **kwargs):
     return decorate
 
 
+def ensure_input_stream_is_not_read(f):
+    """Make sure that the input stream has not been read already."""
+    @wraps(f)
+    def decorate(*args, **kwargs):
+        if request.content_length and request.stream.is_exhausted:
+            abort(500)
```

Comment:
Minor and IMHO: should we maybe add some more clear message? Please improve English! :)
```
abort(500, description="The request stream has been already consumed and cannot be read any more.")
```

Reply:
I am not sure if it makes sense to give an error message on such a low-level detail since the message itself is ambiguous and users won't be able to compose a correct request after reading it...

One could argue that now it is Sentry reading the stream when the content type is `application/json` so we could say "please use `application/octet-stream` instead of `application/json`", but later it could be another library which reads `application/octet-stream` for a different reason and we will be giving a misleading message. Wha do you think? :)

Reply:
I would create a specific exception and raise, this way it's completely clear in Sentry what the issue is:

```python
#errors.py

class ExhausedStreamError(FilesException):
    pass

# views.py
raise ExhausedStreamError()